Source/SwiftLintFramework/Rules/Lint/DuplicateConditionsRule.swift

```diff
@@ -0,0 +1,227 @@
+import SwiftSyntax
+
+struct DuplicateConditionsRule: SwiftSyntaxRule, ConfigurationProviderRule {
+    var configuration = SeverityConfiguration(.warning)
```

Comment:
I wonder if this should be even a error and on by default. Since a duplicated condition in a branching statement usually means one of the branches gets never executed, thus it is in the best case just dead code and in the worst case it is a bug.

Reply:
Makes sense to me --- will change!